src/plugins/intel_cpu/src/nodes/memory.cpp

```diff
@@ -118,57 +235,189 @@ MemoryInput::MemoryInput(const std::shared_ptr<ov::Node>& op, const GraphContext
+    auto selectedPd = getSelectedPrimitiveDescriptor();
+    OPENVINO_ASSERT(selectedPd,
+        "MemoryInput ",
+        getName(),
+        " failed getSelectedPrimitiveDescriptor() call, preferable primitive descriptor is not set");
+
+    auto config = selectedPd->getConfig();
+    config.outConfs.front().setMemDesc(mem_desc);
+    //bypass any checks, we enforce the child descriptor
+    selectedPd->setConfig(config);
+}
+
+void MemoryInput::resolveInPlaceEdges(Edge::LOOK look) {
+    if (!(look & Edge::LOOK_UP)) {
+        Node::resolveInPlaceEdges(look);
+        return;
+    }
+
+    auto selected_pd = getSelectedPrimitiveDescriptor();
+    OPENVINO_ASSERT(selected_pd,
+        "MemoryInput ",
+        getName(),
+        " failed getSelectedPrimitiveDescriptor() call, preferable primitive descriptor is not set");
+
+    auto memDesc = selected_pd->getConfig().outConfs.front().getMemDesc();
+    memMngr = std::make_shared<ProxyMemoryMngr>();
+
+    for (auto&& edge : getChildEdgesAtPort(0)) { // always only one child port
+        OPENVINO_ASSERT(one_of(edge->getStatus(), Edge::Status::Uninitialized, Edge::Status::NotAllocated),
+            " Unexpected inplace resolve call to an allocated edge: ", edge->name());
+
+        auto edgeMem = std::make_shared<Memory>(getEngine(), memDesc, memMngr);
+        edge->reuse(edgeMem);
     }
 }
 
 MemoryInput::~MemoryInput() {
+    if (outputNode) { outputNode->deregisterSibling(this); }
     MemoryNodeVirtualEdge::remove(this, holder);
 }
 
-MemoryPtr MemoryInput::getStore() {
-    return dataStore;
+MemoryOutput& MemoryInput::getOutputNode() {
+    OPENVINO_ASSERT(outputNode, "MemoryOutput ", getName(), " doesn't have sibling input");
+    return *outputNode;
 }
 
-void MemoryInput::storeState(const IMemory &new_state) {
-    // TODO: Should be next one call:
-    //           dataStore.load(new_state, false);
-    //       But because of performance reason we use simple manual copy
-    simple_copy(*dataStore, new_state);
+void MemoryInput::assignState(MemStatePtr newState) {
+    assignedMem = newState->input_mem();
+
+    OPENVINO_ASSERT(assignedMem,
+        "MemoryInput ",
+        getName(),
+        " assigned state has null memory ptr");
+
+    const auto& newDims = assignedMem->getStaticDims();
+    MemoryDescPtr internDesc;
+    if (isDynamicNode()) {
+        const bool hasZeroDims = std::count(std::begin(newDims), std::end(newDims), 0) > 0;
+        internDesc = getBaseMemDescAtOutputPort(0)->cloneWithNewDims(newDims, hasZeroDims);
+    } else {
+        auto expectedDims = getBaseMemDescAtOutputPort(0)->getShape().getStaticDims();
+        OPENVINO_ASSERT(expectedDims == newDims,
+            "MemoryInput ",
+            getName(),
+            " unexpected state shape: ",
+            vec2str(newDims),
+            ", while the expected shape: ",
+            vec2str(expectedDims));
+
+        internDesc = getBaseMemDescAtOutputPort(0);
+    }
+
+    OPENVINO_ASSERT(memMngr,
```

Comment:
Do we expext memMngr is always valid here? How do we make sure MemoryInput::resolveInPlaceEdges is always involked and hence memMngr available?

Reply:
It's always invoked by the graph logic as the node sets inPlace port number different form `-1`. But, to handle any unexpected situations, this assert is there.